lib/common/socket/evloop.c.h

```diff
@@ -399,19 +399,21 @@ static struct st_h2o_evloop_socket_t *create_socket(h2o_evloop_t *loop, int fd,
     return sock;
 }
 
-static struct st_h2o_evloop_socket_t *create_socket_set_nodelay(h2o_evloop_t *loop, int fd, int flags)
+static struct st_h2o_evloop_socket_t *create_socket_set_nodelay(h2o_evloop_t *loop, int fd, int flags, int sa_family)
 {
     /* ignore errors returned by setsockopt; fd is not restricted to TCP sockets (could be a unix-domain socket for example) */
-    int on = 1;
-    setsockopt(fd, IPPROTO_TCP, TCP_NODELAY, &on, sizeof(on));
+    if (sa_family != AF_UNIX) {
+        int on = 1;
+        setsockopt(fd, IPPROTO_TCP, TCP_NODELAY, &on, sizeof(on));
+    }
     return create_socket(loop, fd, flags);
 }
 
 h2o_socket_t *h2o_evloop_socket_create(h2o_evloop_t *loop, int fd, int flags)
 {
     /* it is the reponsibility of the event loop to modify the properties of a socket for its use (i.e., set O_NONBLOCK) */
     fcntl(fd, F_SETFL, O_NONBLOCK);
-    return &create_socket_set_nodelay(loop, fd, flags)->super;
+    return &create_socket_set_nodelay(loop, fd, flags, AF_UNSPEC)->super;
```

Comment:
I think we can:
* call `create_socket` directly here, and
* always call `setsockopt(TCP_NODELAY)`, with a comment stating that we are too lazy to test the socket type